cmd/queue/main.go

```diff
@@ -271,7 +293,13 @@ func probeQueueHealthPath(port int, timeoutSeconds int) error {
 			return false, nil
 		}
 		defer res.Body.Close()
-		return health.IsHTTPProbeReady(res), nil
+		success := health.IsHTTPProbeReady(res)
+		// The check for preferForScaledown() fails readiness faster
+		// in the presence of the label
+		if !success && preferPodForScaledown(env) {
+			cancel()
```

Comment:
I don't fully understand this control flow here. Why cancel() and return false?

Reply:
calling `cancel()` is to short circuit exiting from the `PollImmediateUntil` loop if the label is present. what is returned from the function doesnt matter at that point since canceling and closing the `stopCh` would result in an `ErrWaitTimeout`.